### ISSUE TITLE
Investigate code 500 Internal Server Error when granting

### ISSUE THREAD

**Issue by cygnusv** [nucypher/nucypher#1694]
Lots of nodes are responding to grant requests with the following error:
```
(Ursula)⇀ Foo Bar ↽ (0xfoobar) UnexpectedResponse : Unexpected response while trying to post (),{'data': b'\x03\xf2\xac,\xb2\xce\x82b\xe4R\x12\xa1\xe5z0\x9c+e\x80RdIr_<\xf4\x9bt{\xfb/\x9ae\x02\nv^\xbc79\xf5\xa3s6\xf5\xdf\xd7\xd6\xa1\xedZ&r\x045\xec\xf7+\x03\x82V#\xb2\xbb\xe0N\xa2\x8bQ\x01\xe7\xe7\x82\x9e\xc4\xcb\xd7\xed\xbc=\xab\x85\xbbz\xa7h}\xa3\xb4{\xd1GG\xe4%\x7f\xdbR\x03\xc8r\xb9s\x90\xc1\xa3\xc8\xb5\x1d\x99Q\x9eJ%\xd5\xc4\xee\xb8d\x98x\x1e\x8d\x96`\xb9\xf6\x92\xebe\xc2Y\x83\xc5\xce2 8\xe4,\x1c\xcc\x99\rzC^7n\x84\x17\xdf{u\xb2\xec\xf6o\x95e"\x84\x91\x07\x1d\xaf\x18\xb1\xc6t\xdd\xa1\xb9\xc2\x11y\xf4\xf4\xa8\x84*^\x00\x84\xc9B\x92&*HM\xa4.\xe3\xb9\xbe\xf0\xf1\xbd[Q~T\xe0<\xc7\x94\xfb\x0f\xee\x9e\x95\x14|\xd7\'<\xef}Q\xf7\xeci`P\xa8\xcbe\x97HW:\xa9\x8b\x0f!\xd7\xd1F\x04\x98\x87{p\x92\x84\xc4\xe4\xe8\xb7\x8f\xfb_\xc9\xe4A\xd0\xfbU\xb6\xe4<\xf0*\xb5O\xc8\xf9\xb5\x97+2\xf0\xc0\xbe\xeez\xa7\xbavl\x8ew\xe2V\x01\r\xec\x0c\x06w\xc5\xfd=%\xb8\xaf\x0b\xb4\x13L\x12Ixm\xac\xaa\xffD\xe4\xab\xb4\xacie\x85\x98\x9c\x85\xf1r\x12\x92?\x0cBUq\x18\x9c\x15\xf8\x08\x17\xed\x0c9\x82\xb6|\x1f\xcdH\xa9\xf0_\xcb~\xc13cw\xb7M}\xa1&&\xc3\xfa@\xa1VMd\x9d}Y\xda\xddw>\xe2\xee\xf2\x87\x18z\xc5\x9b\x8e\xb1\xcd\x92\xa3\x87x;#c\x08\xfa\x903\xc7-\xfc|\xab\xae\xb11\x17g\x0cJ-\xadEu\x18_\xa9\xf8\x05\xdb\x8c\x10\x8b\xa1\x97r\x9d\xdc\x7f\xe4\x93\xc8\x8c\xb3\x87\xf6\xbc\x85i\x18\xbd!\xf2\x1b\x10\x85\xc4\xad*\'\x1b\xa5r\x11.U\xa8\x03I\xa3\xaatD\xbf\xec\xc8L\xbdXL\xecZE\x1f_\x86\xd9\xac\xc6\xd7\x84\x7f\xf5\xa8\xbb=\'\x01\x10\xb7\x8d\x86\xaa\x8b', 'timeout': 2}: 500 b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<title>500 Internal Server Error</title>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>\n'
```

**Comment by tuxxy** [comment]
Have you found a way to reproduce?

**Comment by KPrasch** [comment]
This appears to be an issue with treasure map distribution?

**Comment by KPrasch** [comment]
Improves Ursula response codes during consider arrangement; Fixed in #2562
